src/ts/Detail/Comments.tsx

```diff
@@ -4,66 +4,130 @@ import {
   Paper,
 } from 'material-ui';
 import * as _ from 'lodash';
+import comments from './comments-mockup';
 
-export default function Comments(props) {
-  const stars = _.times(5, i => <FontIcon key={i} className="fa fa-star"/>);
-  return (
-    <div className="comments">
-      <Paper style={{ width: '100%' }} zDepth={2} rounded={false}>
-        <div className="comments-cont">
-          <div className="references">
-            55 References
-          </div>
-          <div className="ratings">
-            <div className="row">
-              <div className="rate">
-                <div className="type">Bike performance</div>
-                <div className="stars">{stars}</div>
-              </div>
-              <div className="rate">
-                <div className="type">Communication</div>
-                <div className="stars">{stars}</div>
-              </div>
-            </div>
-            <div className="row">
-              <div className="rate">
-                <div className="type">Mechanical condition</div>
-                <div className="stars">{stars}</div>
-              </div>
-              <div className="rate">
-                <div className="type">Rating</div>
-                <div className="stars">{stars}</div>
-              </div>
+export default class Comments extends React.Component<any, any> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      offset: 0,
+      limit: 2,
+      loading: true,
+      fetching: false,
+      comments: [],
+    }
+    this.comment = this.comment.bind(this);
+    this.rating = this.rating.bind(this);
+    this.fetch = this.fetch.bind(this);
+  }
+
+  componentDidMount() { this.fetch(); }
+
+  fetch() {
+    this.setState({ fetching: true });
+    setTimeout(() => {
+      this.setState({
+        loading: false,
+        fetching: false,
+        rating: comments.rating,
+        references: comments.references,
+        comments: [
+          ...this.state.comments,
+          ...comments.comments,
+        ],
+      });
+    }, 2000);
+  }
+
+  render() {
+    if (this.state.loading) {
+      return (
+        <div className="comments">
+          <Paper style={{ width: '100%' }} zDepth={2} rounded={false}>
+            <div className="comments-cont">
+              <div style={{textAlign: 'center'}}>LOADING COMMENTS...</div>
             </div>
+          </Paper>
+        </div>
+      );
+    }
+
+    const rating = this.rating();
+    const comments = this.comment();
+    const more = this.state.fetching ? <div>Loading comments</div> : <div onClick={this.fetch}>View all references</div>;
+    return (
+      <div className="comments">
+        <Paper style={{ width: '100%' }} zDepth={2} rounded={false}>
+          <div className="comments-cont">
+            <div className="references">{this.state.references} References</div>
```

Comment:
the two returns seem only to differ with the contents of `.comments-cont` – let's keep it DRY